### PR TITLE
Export de-components Lib and Minor refactor. 

### PR DESCRIPTION
**Pull request by sri2k1us**
- As part of my experiments, I also added a `Rate` component.
See https://github.com/sri2k1us/de-components/pull/1 for more info.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,11 +7,11 @@
     "@material-ui/icons": "^3.0.2",
     "react": "^16.8.3",
     "react-dom": "^16.8.3",
+    "react-rating": "^1.6.2",
     "react-scripts": "2.1.5"
   },
   "scripts": {
-    "start": "react-scripts start",
-    "build": "react-scripts build",
```
Comment (psarando) [inline package.json:14]: I wonder if there's a way to accomplish this with just `react-scripts build`? Maybe if everything is still exported from `src/index.js`? 🤔 

I can try experimenting with it before or after this PR is merged.

File: src/components/announcer/CyverseAnnouncer.js [hunk 2]
```diff
@@ -1,10 +1,9 @@
 import React, { Component } from 'react';
 import Announcer from "./Announcer";
-import PropTypes from "prop-types";
 
 
 let msgQueue = [];
-const timeout = 6000;
+const TIMEOUT = 6000;
```
Comment (psarando) [inline src/components/announcer/CyverseAnnouncer.js]: 👍

File: src/components/announcer/CyverseAnnouncer.js [hunk 3]
```diff
@@ -29,24 +28,30 @@ class CyverseAnnouncer extends Component {
   }
 
   componentDidMount() {
-    let timer = setInterval(this.tickCallback, timeout);
+    let timer = setInterval(this.tickCallback, TIMEOUT);
+    //display first message right away
+    this.dequeue();
```
Comment (psarando) [inline src/components/announcer/CyverseAnnouncer.js]: 👍

File: src/components/announcer/CyverseAnnouncer.js [hunk 4]
```diff
@@ -55,26 +60,27 @@ class CyverseAnnouncer extends Component {
...
 
-CyverseAnnouncer.propTypes = {
-  message: PropTypes.string,
-  variant: PropTypes.oneOf(['success', 'warning', 'error', 'info']).isRequired,
-  duration: PropTypes.number,
-  vertical: PropTypes.oneOf(['top', 'bottom']),
-  horizontal: PropTypes.oneOf(['center', 'right', 'left']),
+/**
+ *  Queue messages needed to be announced using CyverseAnnouncer
+ *  @param {{text: string, variant: string, duration: number, horizontal: string, vertical: string}} msg
```
Comment (psarando) [inline src/components/announcer/CyverseAnnouncer.js]: We should complete this documentation by further documenting this object:
```javascript
/**
 * @typedef {(TOP|BOTTOM)} VerticalAlignmentEnum
 */

/**
 * @typedef {(RIGHT|LEFT|CENTER)} HorizontalAlignmentEnum
 */

/**
 * @typedef {(INFO|ERROR|WARNING|SUCCESS)} VariantEnum
 */

/**
 * @typedef {Object} QueueMessage
 * @property {string} text - The message text.
 * @property {VariantEnum} variant - The message variant.
 * @property {number} duration - The message duration.
 * @property {HorizontalAlignmentEnum} horizontal - The message horizontal alignment.
 * @property {VerticalAlignmentEnum} vertical - The message vertical alignment.
 */

/**
 *  Queue messages needed to be announced using CyverseAnnouncer
 *  @param {QueueMessage} msg - A message with configuration.
 */
```

Then above, these enum constants can also be documented:
```javascript
/**
 * A message variant enum that will display info messages.
 * @constant
 */
const INFO = "info";
```

File: src/components/lib.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import CyverseAnnouncer, { announce } from "./announcer/CyverseAnnouncer";
+import Rate from "./rating/Rate";
+
+export {
+    CyverseAnnouncer,
+    announce,
+    Rate,
```
Comment (psarando) [inline src/components/lib.js]: They way we have the `build` script and this export set up now, I think we would have to use something like the following in our main UI code to use these components:
```javascript
import { CyVerseAnnouncer, announce, Rate } from "@cyverse-de/de-components";
```

I wonder if we should imitate a package structure similar to [material-ui](https://github.com/mui-org/material-ui/blob/next/packages/material-ui/src/index.js) (and we should decide soon before these components are imported in too many places)?

For example, to use a MUI `Button` and an icon:
```javascript
import Button from "@material-ui/core/Button";
import HelpIcon from "@material-ui/icons/Help";
```

Would it be more convenient to use imports like the following for our components?
```javascript
import CyVerseAnnouncer from "@cyverse-de/de-components/CyVerseAnnouncer";
import Rate from "@cyverse-de/de-components/Rate";
import SomethingSoon from "@cyverse-de/de-components/future-sub-package/SomethingSoon";
```

Basically treating `@cyverse-de/de-components` the same as `@material-ui/core`, but sub-packaging from there rather than creating separate packages (like `@material-ui/icons`).

I don't know the best way to organize the exports to accomplish this right now, but we could experiment and figure something out if we wanted to go this way.

File: src/components/lib.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import CyverseAnnouncer, { announce } from "./announcer/CyverseAnnouncer";
```
Comment (psarando) [inline src/components/lib.js]: I think `CyVerse` should be properly cased, so at least this import should be renamed `CyVerseAnnouncer`, but the file and classes in `CyverseAnnouncer.js` should probably get renamed to `CyVerseAnnouncer` as well.

File: .gitignore [hunk 7]
```diff
@@ -23,3 +25,7 @@ yarn-debug.log*
 yarn-error.log*
 
 test-report.xml
+*.tgz
+dist/lib.js
```
Comment (psarando) [inline .gitignore]: Since we already ignore `/dist` above, we don't need this `dist/lib.js` entry too.

File: src/components/rating/Rate.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+                    emptySymbol={<img src={whitestar} className="icon" alt="white star"/>}
+                    fullSymbol={<img src={goldstar} className="icon" alt="gold star"/>}
+                    placeholderSymbol={<img src={redstar} className="icon"
+                                            alt="red star"/>}
+                    fractions={2}
+                    readonly={true}
+                />
+                <span style={{paddingLeft: 3}}>
+                                ({total})
+                        </span>
```
Comment (psarando) [inline src/components/rating/Rate.js]: Looks like one of these `span` tags is at the wrong indentation level.

**Review by psarando — CHANGES_REQUESTED** [review]
Looks good 👍 

I have some questions on the library organization, and a few suggestions, for docs, branding, etc.

File: package.json (hunk 1 above)
Comment (psarando) [reply package.json:14]: Oh wait, never mind that. Since `react-scripts build` generates a `build` folder with `index.html` and assets under `build/static/js` that have cache-busting names. So that won't work for library components.

File: src/components/lib.js (hunk 5 above)
Comment (ash3rz) [reply src/components/lib.js]: I think it'd be more convenient to have the imports like how MUI has them, but also not sure how much work it would take to replicate MUI's package structure.

**Review by ash3rz — COMMENTED** [review]
I had the same feeling about using proper casing for `CyVerse` that Paul mentioned, but other than that, LGTM 👍

File: src/components/announcer/CyverseAnnouncer.js (hunk 4 above)
Comment (sri2k1us) [reply src/components/announcer/CyverseAnnouncer.js]: Great! I didn't knew how to document Objects. Thanks

File: src/components/lib.js (hunk 6 above)
Comment (sri2k1us) [reply src/components/lib.js]: Sounds good. I will rename them.

File: .gitignore (hunk 7 above)
Comment (sri2k1us) [reply .gitignore]: 👍

File: src/components/rating/Rate.js (hunk 8 above)
Comment (sri2k1us) [reply src/components/rating/Rate.js]: Prettier - Please take care of it 😉

**Comment by sri2k1us** [comment]
I have run `prettier` once in this branch.

File: src/App.js [hunk 9]
```diff
@@ -1,23 +1,26 @@
...
+                <header className="App-header">
+                    <p>
+                        <a
+                            className="App-link"
+                            target="_blank"
+                            rel="noopener noreferrer"
+                            href="https://cyverse-de.github.io/de-components/"
+                        >
+                            {" "}
+                            DE Components{" "}
```
Comment (psarando) [inline src/App.js]: LGTM, :shipit: once you delete these extra spaces 👾

**Review by psarando — APPROVED** [review]

**Comment by sri2k1us** [comment]
I updated version in `package.json` with `npm version minor -m 'Upgrade to %s'`.

**Comment by sri2k1us** [comment]
Thanks for the reviews!
